### PR TITLE
Updated the yaml to reflect the checks that both count and page parameters are positive numbers

### PR DESCRIPTION
**Pull request by olakunlekuye2**

### CODE REVIEW

**Comment by drnaylor-oc** [comment]
This PR is in a slightly weird state as it has some commits that have been merged -- which it shouldn't be displaying? Anyhow -- just basing my comments on the last commit here. I'm not criticising what's been done here, just thinking out loud -- and trying to make sure every PR that is currently out has had some review.

Are the changes to the definition here necessary -- that is, should we assume that developers should know this implicitly? Or should we explicitly declare this. Should we update the examples for 400s instead to include the bad request error with the specific text for non-positive page/count, or is that superfluous?

Do we need to do any of this?

@fmcgrath do you have any thoughts on the above? You'll just want to look at https://github.com/hmrc/common-transit-convention-traders/pull/449/commits/56bb5ab63ac6e4adaec3a818f2df88c12e63a92b.

File: resources/public/api/conf/2.0/application.yaml [hunk 1]
```diff
@@ -165,14 +165,18 @@ paths:
             example: 3CnsTh79I7vtOy6
         - in: query
           name: page
-          description: This is an optional query parameter, default is 1, that allows the selection of a specific page from a multi-page query.
+          description: >
+            This is an optional query parameter, default is 1, that allows the selection of a specific page from a multi-page query.
+            A check is made on the page parameter to ensure it is a positive number.
```
Comment (fmcgrath) [inline resources/public/api/conf/2.0/application.yaml:170]: Suggestion:
This is an optional query parameter (default value is `1`) that allows the selection of a specific page from a multi-page query. If used, you must enter a positive number.

File: resources/public/api/conf/2.0/application.yaml [hunk 2]
```diff
@@ -379,14 +383,18 @@ paths:
             example: 2021-06-21T09:00:00Z
         - in: query
           name: page
-          description: This is an optional query parameter, default is 1, that allows the selection of a specific page from a multi-page query.
+          description: >
+            This is an optional query parameter, default is 1, that allows the selection of a specific page from a multi-page query.
+            A check is made on the page parameter to ensure it is a positive number.
```
Comment (fmcgrath) [inline resources/public/api/conf/2.0/application.yaml:388]: Suggestion:
This is an optional query parameter (default value is `1`) that allows the selection of a specific page from a multi-page query. If used, you must enter a positive number.

File: resources/public/api/conf/2.0/application.yaml [hunk 3]
```diff
@@ -906,14 +914,18 @@ paths:
             example: 3CnsTh79I7vtOy6
         - in: query
           name: page
-          description: This is an optional query parameter, default is 1, that allows the selection of a specific page from a multi-page query.
+          description: >
+            This is an optional query parameter, default is 1, that allows the selection of a specific page from a multi-page query.
+            A check is made on the page parameter to ensure it is a positive number.
```
Comment (fmcgrath) [inline resources/public/api/conf/2.0/application.yaml:919]: Suggestion:
This is an optional query parameter (default value is `1`) that allows the selection of a specific page from a multi-page query. If used, you must enter a positive number.

File: resources/public/api/conf/2.0/application.yaml [hunk 4]
```diff
@@ -1127,14 +1139,18 @@ paths:
             example: 2021-06-21T09:00:00Z
         - in: query
           name: page
-          description: This is an optional query parameter, default is 1, that allows the selection of a specific page from a multi-page query.
+          description: >
+            This is an optional query parameter, default is 1, that allows the selection of a specific page from a multi-page query.
+            A check is made on the page parameter to ensure it is a positive number.
```
Comment (fmcgrath) [inline resources/public/api/conf/2.0/application.yaml:1144]: Suggestion:
This is an optional query parameter (default value is `1`) that allows the selection of a specific page from a multi-page query. If used, you must enter a positive number.

File: resources/public/api/conf/2.0/application.yaml [hunk 5]
```diff
@@ -165,14 +165,18 @@ paths:
...
           style: form
           schema:
             type: integer
             example: 2
         - in: query
           name: count
-          description: This is an optional query parameter, default is 25, that allows the selection of a the number of items per page in a query.
+          description: >
+            This is an optional query parameter, default is 25, that allows the selection of a the number of items per page in a query.
+            A check is made on the count parameter to ensure it is a positive number.
```
Comment (fmcgrath) [inline resources/public/api/conf/2.0/application.yaml:179]: Suggestion:
This is an optional query parameter (default value is `25`) that allows the selection of the number of items per page in a multi-page query. If used, you must enter a positive number.

File: resources/public/api/conf/2.0/application.yaml [hunk 6]
```diff
@@ -379,14 +383,18 @@ paths:
...
           style: form
           schema:
             type: integer
             example: 2
         - in: query
           name: count
-          description: This is an optional query parameter, default is 25, that allows the selection of the number of items per page in a multi-page query.
+          description: >
+            This is an optional query parameter, default is 25, that allows the selection of the number of items per page in a multi-page query.
+            A check is made on the count parameter to ensure it is a positive number.
```
Comment (fmcgrath) [inline resources/public/api/conf/2.0/application.yaml:397]: Suggestion:
This is an optional query parameter (default value is `25`) that allows the selection of the number of items per page in a multi-page query. If used, you must enter a positive number.

File: resources/public/api/conf/2.0/application.yaml [hunk 7]
```diff
@@ -906,14 +914,18 @@ paths:
...
           style: form
           schema:
             type: integer
             example: 2
         - in: query
           name: count
-          description: This is an optional query parameter, default is 25, that allows the selection of a the number of items per page in a multi-page query.
+          description: >
+            This is an optional query parameter, default is 25, that allows the selection of a the number of items per page in a multi-page query.
+            A check is made on the count parameter to ensure it is a positive number.
```
Comment (fmcgrath) [inline resources/public/api/conf/2.0/application.yaml:928]: Suggestion:
This is an optional query parameter (default value is `25`) that allows the selection of the number of items per page in a multi-page query. If used, you must enter a positive number.

File: resources/public/api/conf/2.0/application.yaml [hunk 8]
```diff
@@ -1127,14 +1139,18 @@ paths:
...
           style: form
           schema:
             type: integer
             example: 2
         - in: query
           name: count
-          description: This is an optional query parameter, default is 25, that allows the selection of the number of items per page in a multi-page query.
+          description: >
+            This is an optional query parameter, default is 25, that allows the selection of the number of items per page in a multi-page query.
+            A check is made on the count parameter to ensure it is a positive number.
```
Comment (fmcgrath) [inline resources/public/api/conf/2.0/application.yaml:1153]: Suggestion:
This is an optional query parameter (default value is `25`) that allows the selection of the number of items per page in a multi-page query. If used, you must enter a positive number.

**Review by fmcgrath — CHANGES_REQUESTED** [review]
Some suggestions for rewording.

**Comment by drnaylor-oc** [comment]
Can you rebase your commits on top of the current main please? I'm a little concerned about my commits being there which should also be in main, yet Github hasn't picked them up (not your fault, but I don't want something to go wrong when merging).

File: resources/public/api/conf/1.0/application.yaml [hunk 9]
```diff
@@ -1224,20 +1382,38 @@ paths:
...
+                  - $ref: '#/components/schemas/NotFound'
+                  - description: No box found for your client id
+                    example:
+                      statusCode: 404
+                      message: No box found for your client id
+                      code: CLIENT_ERROR
+              example:
+                statusCode: 404
+                message: No box found for your client id
+                code: CLIENT_ERROR
```
Comment (fmcgrath) [inline resources/public/api/conf/1.0/application.yaml:1416]: In this block, `client id` is used but in preceding block `Client Id` is used. We need to be more consistent.

File: resources/public/api/conf/1.0/application.yaml [hunk 10]
```diff
@@ -1224,20 +1382,38 @@ paths:
...
+        '400':
+          description: Bad Request
           content:
             application/json:
               schema:
                 allOf:
-                  - $ref: '#/components/schemas/NotAcceptable'
-                  - description: The Accept header is missing or invalid.
+                  - $ref: '#/components/schemas/BadRequest1'
+                  - description: Client Id Required.
```
Comment (fmcgrath) [inline resources/public/api/conf/1.0/application.yaml:1392]: Use lowercase for "required". In general, we are using sentence case.

**Comment by drnaylor-oc** [comment]
Version 1 shouldn't even be in here, that'll be from previous commits.

@olakunlekuye2 can you just take your version 2 yaml file and put it on a new branch from main, then close this PR?

File: resources/public/api/conf/1.0/application.yaml [hunk 11]
```diff
@@ -1270,24 +1446,43 @@ components:
         message:
           type: string
           description: The description of the error.
-    NotFound:
-      title: NotFound
-      description: Supplied arrival/departure not found in the database or does not exist.
+    BadRequest1:
+      title: BadRequest1
+      description: An error in the supplied payload.
```
Comment (fmcgrath) [inline resources/public/api/conf/1.0/application.yaml:1451]: Would `Error in the supplied payload` be better?

**Review by fmcgrath — COMMENTED** [review]
I skimmed through the changes in the two API definition files. No major issues.

**Pull request closed without merging**